File: src/main/java/de/axelrindle/chickendropfeathersmod/EntitySpawnHandler.java
package de.axelrindle.chickendropfeathersmod;

import net.minecraft.entity.Entity;
import net.minecraft.entity.passive.EntityChicken;
import net.minecraftforge.event.entity.EntityJoinWorldEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;

public class EntitySpawnHandler {

    @SubscribeEvent
    public void onEntitySpawn(EntityJoinWorldEvent event) {
        Entity entity = event.getEntity();
        if (entity instanceof EntityChicken) {
            EntityChicken chicken = ((EntityChicken) entity);
            chicken.tasks.addTask(
                    chicken.tasks.taskEntries.size(),
                    new EntityAIDropFeather(chicken)
            );
        }
    }
}

File: src/main/kotlin/de/axelrindle/chickendropfeathersmod/EntitySpawnHandler.kt
package de.axelrindle.chickendropfeathersmod

import net.minecraft.entity.passive.EntityChicken
import net.minecraftforge.event.entity.EntityJoinWorldEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

class EntitySpawnHandler {

    @SubscribeEvent
    fun onEntitySpawn(event: EntityJoinWorldEvent) {
        val entity = event.entity
        if (entity is EntityChicken)
            entity.tasks.addTask(
                    entity.tasks.taskEntries.size,
                    EntityAIDropFeather(entity)
            )
    }
}